### PR TITLE
Update integration tests to verify the open sockets on system level - Closes #1910

### LINKED ISSUE #1910: Update integration tests to verify the open sockets on system level
### Expected behavior

During integration tests we expect the desired number of socket connections opened, based on fact the number of instances we start and connect to each other. 

### Actual behavior
We noticed huge amount of socket connections opened on system level, which are not expected to be there. 

```
tcp        0      0 159.65.154.233:47136    167.99.166.85:5000      ESTABLISHED
tcp        0      0 159.65.154.233:47902    206.189.74.122:5000     ESTABLISHED
tcp        0      0 159.65.154.233:37552    37.139.22.164:5000      ESTABLISHED
tcp        0      0 159.65.154.233:48546    104.236.8.170:5000      ESTABLISHED
tcp        0      0 159.65.154.233:35096    159.89.118.25:5000      ESTABLISHED
tcp        0      0 159.65.154.233:35444    206.189.70.89:5000      ESTABLISHED
tcp        4      0 159.65.154.233:52872    159.65.119.184:5000     ESTABLISHED
tcp        0      0 159.65.154.233:38186    206.189.86.230:5000     ESTABLISHED
tcp        0      0 159.65.154.233:47216    167.99.144.31:5000      ESTABLISHED
tcp        0      0 159.65.154.233:40004    206.189.83.233:5000     ESTABLISHED
tcp        0      0 159.65.154.233:59356    104.236.41.19:5000      ESTABLISHED
tcp        0      0 159.65.154.233:56736    206.189.10.16:5000      ESTABLISHED
tcp        0      0 159.65.154.233:58960    206.189.30.207:5000     ESTABLISHED
tcp        0      0 159.65.154.233:36776    188.166.100.49:5000     ESTABLISHED
tcp        0      8 159.65.154.233:37158    104.236.12.16:5000      ESTABLISHED
tcp        0    528 159.65.154.233:47416    104.236.12.16:5000      ESTABLISHED
tcp        0      0 159.65.154.233:52010    206.189.83.79:5000      ESTABLISHED
tcp        4      0 159.65.154.233:59150    167.99.227.32:5000      ESTABLISHED
tcp        0      0 159.65.154.233:53606    206.189.83.227:5000     ESTABLISHED
tcp        0      0 159.65.154.233:59022    146.185.142.247:5000    ESTABLISHED
tcp        0      0 159.65.154.233:60230    159.89.123.55:5000      ESTABLISHED
tcp        0      0 159.65.154.233:48042    146.185.142.199:5000    ESTABLISHED
tcp        0      0 159.65.154.233:56034    82.196.13.80:5000       ESTABLISHED
tcp        4      0 159.65.154.233:60618    206.189.83.229:5000     ESTABLISHED
tcp        0      0 159.65.154.233:48518    206.189.30.105:5000     ESTABLISHED
tcp        0      0 159.65.154.233:57924    167.99.42.72:5000       ESTABLISHED
tcp        0      0 159.65.154.233:33888    159.65.157.22:5000      ESTABLISHED
tcp        4      0 159.65.154.233:59062    146.185.142.106:5000    ESTABLISHED
tcp        0      0 159.65.154.233:52514    206.189.16.145:5000     ESTABLISHED
tcp        0      0 159.65.154.233:49276    104.236.40.25:5000      ESTABLISHED
tcp        0      0 159.65.154.233:50782    138.197.130.102:5000    ESTABLISHED
tcp        0      0 159.65.154.233:57302    206.189.81.168:5000     ESTABLISHED
tcp        4      0 159.65.154.233:39524    159.203.42.44:5000      ESTABLISHED
tcp        0      0 159.65.154.233:37180    206.189.81.168:5000     ESTABLISHED
tcp        0      0 159.65.154.233:38334    167.99.166.85:5000      ESTABLISHED
tcp        4      0 159.65.154.233:51444    146.185.142.78:5000     ESTABLISHED
tcp        0      0 159.65.154.233:50600    206.189.70.69:5000      ESTABLISHED
tcp        4      0 159.65.154.233:37278    104.236.37.186:5000     ESTABLISHE
```

### Steps to reproduce

On any devnet machine run this command `lsof -i :port`

### Which version(s) does this affect? (Environment, OS, etc...)
`1.0.0`

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

There were unexpected number of web socket connections on a running node. And we were not checking those during integration tests. 

### How did I fix it?
Added some test scenarios to check the open ports. 

### How to test it?

Run integration tests 

### Review checklist

* The PR solves #1910
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/integration/utils/index.js [hunk 1]
```diff
@@ -24,4 +25,37 @@ module.exports = {
...
+		childProcess.exec(
+			`lsof ${ports
+				.map(p => `-i :${p}`)
+				.join(' ')} -P -n -s TCP:LISTEN -t | wc -l`,
+			(err, stdout) => {
+				cb(err, parseInt(stdout.toString().trim()));
+			}
+		);
+	},
+	getOutgoingConnections: (ports, cb) => {
```
Comment (jondubois) [inline test/integration/utils/index.js]: Should this one be named `getEstablishedConnections`?

Comment (nazarhussain) [reply test/integration/utils/index.js]: Already renamed.

**Comment by nazarhussain** [comment]
For the purpose of review, these are the list of open ports while 10 peers connected to each other. 

```
127.0.0.1:56204->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56205->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56206->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56412->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56413->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56414->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56416->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56417->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56418->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56207->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56208->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56420->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56421->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56422->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56423->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56424->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56425->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56426->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56194->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56202->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56403->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56404->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56405->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56406->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56407->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56408->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56409->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56219->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56222->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56224->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56226->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56402->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56458->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56442->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56459->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56460->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56211->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56213->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56214->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56432->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56433->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56434->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56435->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56452->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56453->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56225->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56253->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56446->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56454->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56455->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56456->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56457->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56327->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56398->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56215->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56216->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56217->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56410->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56437->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56438->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56439->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56461->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56463->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56223->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56227->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56395->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56401->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56443->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56444->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56462->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56464->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56465->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56209->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56210->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56212->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56411->127.0.0.1:5009 (ESTABLISHED)
127.0.0.1:56428->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56429->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56430->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56450->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56451->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:56359->127.0.0.1:5007 (ESTABLISHED)
127.0.0.1:56396->127.0.0.1:5008 (ESTABLISHED)
127.0.0.1:56397->127.0.0.1:5003 (ESTABLISHED)
127.0.0.1:56399->127.0.0.1:5002 (ESTABLISHED)
127.0.0.1:56400->127.0.0.1:5000 (ESTABLISHED)
127.0.0.1:56448->127.0.0.1:5006 (ESTABLISHED)
127.0.0.1:56466->127.0.0.1:5005 (ESTABLISHED)
127.0.0.1:56467->127.0.0.1:5004 (ESTABLISHED)
127.0.0.1:56468->127.0.0.1:5001 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56206 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56208 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56212 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56214 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56217 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56222 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56223 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56398 (ESTABLISHED)
127.0.0.1:5002->127.0.0.1:56399 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56202 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56207 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56216 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56253 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56395 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56400 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56432 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56450 (ESTABLISHED)
127.0.0.1:5000->127.0.0.1:56458 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56404 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56414 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56451 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56453 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56457 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56460 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56463 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56465 (ESTABLISHED)
127.0.0.1:5001->127.0.0.1:56468 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56462 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56224 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56327 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56403 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56417 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56421 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56430 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56461 (ESTABLISHED)
127.0.0.1:5004->127.0.0.1:56467 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56359 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56410 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56194 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56413 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56424 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56429 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56434 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56442 (ESTABLISHED)
127.0.0.1:5007->127.0.0.1:56454 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56204 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56210 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56227 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56408 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56423 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56433 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56448 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56456 (ESTABLISHED)
127.0.0.1:5006->127.0.0.1:56459 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56215 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56226 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56396 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56406 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56418 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56425 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56452 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56455 (ESTABLISHED)
127.0.0.1:5008->127.0.0.1:56464 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56205 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56219 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56401 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56407 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56411 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56426 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56435 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56439 (ESTABLISHED)
127.0.0.1:5009->127.0.0.1:56446 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56209 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56213 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56225 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56397 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56405 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56416 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56420 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56437 (ESTABLISHED)
127.0.0.1:5003->127.0.0.1:56443 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56211 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56402 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56409 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56412 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56422 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56428 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56438 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56444 (ESTABLISHED)
127.0.0.1:5005->127.0.0.1:56466 (ESTABLISHED)
```

**Comment by nazarhussain** [comment]
@jondubois As per discussion with @MaciejBaj skipped the failing test in this PR and will look into it under the issue https://github.com/LiskHQ/lisk/issues/1954

File: test/integration/scenarios/network/peers.js [hunk 2]
```diff
@@ -154,6 +154,20 @@ module.exports = function(params) {
 					return expect(networkAverageHeight).to.be.above(1);
 				});
 
+				it('should have valid values values matching specification', () => {
+					return getAllPeers().then(results => {
+						return results.map(peersList => {
+							return peersList.peers.map(peer => {
+								expect(peer.ip).to.not.empty;
+								expect(peer.wsPort).to.be.gte(5000);
+								expect(peer.wsPort).to.be.lt(5010);
```
Comment (MaciejBaj) [inline test/integration/scenarios/network/peers.js]: httpPort - then I expect it to fail, correct?

File: test/integration/transport.js [hunk 3]
```diff
@@ -95,6 +117,42 @@ describe('given configurations for 10 nodes with address "127.0.0.1", WS ports 5
 					});
 				});
 
+				it('there should exactly 10 listening connections for 500[0-9] ports', done => {
```
Comment (MaciejBaj) [inline test/integration/transport.js]: 10 or 9, it doesn't correlate with the expectation of 180 in the test below.

File: test/integration/transport.js [hunk 4]
```diff
@@ -95,6 +117,42 @@ describe('given configurations for 10 nodes with address "127.0.0.1", WS ports 5
...
+							done();
+						} else {
+							done(
+								`There are ${numOfConnections} listening connections on web socket ports.`
+							);
+						}
+					});
+				});
+
+				it('there should maximum 180 established connections from 500[0-9] ports', done => {
```
Comment (MaciejBaj) [inline test/integration/transport.js]: Please change the hardcoded numbers to rely on the introduced `totalPeers` constant.
like `[totalPeer * 9]`.

File: test/integration/scenarios/network/peers.js (hunk 2 above)
Comment (nazarhussain) [reply test/integration/scenarios/network/peers.js]: Yes but `rpc.list` never return the `httpPort`

File: test/integration/transport.js (hunk 3 above)
Comment (nazarhussain) [reply test/integration/transport.js]: There should be 10 listening connections as we started 10 instances.

**Pull request merged**